src/main.rs

```diff
@@ -144,32 +145,48 @@ fn launch_git_with_self_as_editor() {
     run_cmd(Command::new("git").arg("commit").env("GIT_EDITOR", self_path))
 }
 
-fn collect_information_and_write_to_file(out_path: PathBuf) {
-    let maybe_emoji = select_emoji();
-
-    if maybe_emoji == None {
-        abort();
+fn git_message_is_empty(git_commit_contents: & String) -> bool {
```

Comment:
`&String` is almost never what you want to do. I couldn't find a great resource on this, but here is some discussion: https://users.rust-lang.org/t/whats-the-difference-between-string-and-str/10177

```suggestion
fn git_message_is_empty(git_commit_contents: &str) -> bool {
```

I can recommend running [clippy](https://github.com/rust-lang/rust-clippy) which would warn you here:

<img width="599" alt="Screenshot 2022-10-11 at 19 17 16" src="https://user-images.githubusercontent.com/189580/195145589-c0228eb2-c9b3-4ad3-bceb-3e8e1c3918f9.png">